qiskit_experiments/framework/composite/composite_experiment_data.py

```diff
@@ -103,3 +113,56 @@ def _add_single_data(self, data):
                     else:
                         sub_data["counts"] = data["counts"]
                 self._components[index].add_data(sub_data)
+
+    def save(self, verbose=True) -> None:
+        super().save(verbose)
+        for comp in self._components:
+            comp.save(verbose=False)
+
+    def save_metadata(self) -> None:
+        super().save_metadata()
+        for comp in self._components:
+            comp.save_metadata()
+
+    @classmethod
+    def load(cls, experiment_id: str, service: DatabaseServiceV1) -> "CompositeExperimentData":
+        expdata = DbExperimentDataV1.load(experiment_id, service)
+        components = []
+        for comp_id, comp_class in zip(
+            expdata.metadata["component_ids"], expdata.metadata["component_classes"]
+        ):
+            load_class = globals()[comp_class]
+            load_func = getattr(load_class, "load")
+            loaded_comp = load_func(comp_id, service)
+            components.append(loaded_comp)
+
+        expdata.__class__ = CompositeExperimentData
+        expdata._experiment = None
+        expdata._components = components
+
+        return expdata
+
+    def _set_service(self, service: DatabaseServiceV1) -> None:
+        """Set the service to be used for storing experiment data.
+
+        Args:
+            service: Service to be used.
+
+        Raises:
+            DbExperimentDataError: If an experiment service is already being used.
+        """
+        DbExperimentDataV1._set_service(self, service)
+        for comp in self._components:
+            comp._set_service(service)
+
+    def _set_share_level_without_save(self, new_level: str) -> None:
```

Comment:
I think this would be cleaner as override of the `share_level` setter as:
```python
@ExperimentData.share_level.setter
def share_level(self, new_level: str) -> None:
    self._share_level = new_level
    for comp in self._components:
        comp._share_level = new_level
    if self.auto_save:
        self.save_metadata()

Reply:
I see, so this is to work with case of arbitrary nesting of composite experiments. 

I think the recursive saving would only happen for `comp.share_level = share_level` if you manually set `auto_save = True` for sub experiment data. Currently I think they should default to false when constructed within a composite data. So if you only set autosave on the parent experiment, and its save call recursively saves sub experiments, this might not be an issue.

Reply:
I don't think thats much of an issue. Since comp exp data .auto_save doesn't recursively set auto save to sub experiments someone would have to intentionally set auto save = True to a subexperiment. If someone really wants to do this then everything will still work, but some methods will be a bit slower / redundant because of recursive save, but I don't think thats anything we really need to worry about since it is not normal use, so better to have cleaner methods instead.